src/solvers/cvc/cvc_conv.cpp

```diff
@@ -58,14 +58,552 @@ Function: cvc_convt::l_get
+      convert_expr(op);
+    else if(from_width<to_width)
+    {
+      out << "SX(";
+      convert_expr(op);
+      out << ", " << to_width << ")";
+    }
+    else
+    {
+      out << "(";
+      convert_expr(op);
+      out << ")[" << (to_width-1) << ":0]";
+    }
+  }
+  else if(op.type().id()==ID_unsignedbv)
+  {
+    unsigned from_width=
+      unsafe_string2unsigned(id2string(op.type().get(ID_width)));
+
+    if(from_width==to_width)
+      convert_expr(op);
+    else if(from_width<to_width)
+    {
+      out << "(0bin";
+
+      for(unsigned i=from_width; i<to_width; i++)
+        out << "0";
+
+      out << " @ ";
+
+      out << "(";
+      convert_expr(op);
+      out << "))";
+    }
+    else
+    {
+      out << "(";
+      convert_expr(op);
+      out << ")[" << (to_width-1) << ":0]";
+    }
+  }
+  else if(op.type().id()==ID_bool)
+  {
+    if(to_width>1)
+    {
+      out << "(0bin";
+
+      for(unsigned i=1; i<to_width; i++)
+        out << "0";
+
+      out << " @ ";
+
+      out << "IF ";
+      convert_expr(op);
+      out << " THEN 0bin1 ELSE 0bin0 ENDIF)";
+    }
+    else
+    {
+      out << "IF ";
+      convert_expr(op);
+      out << " THEN 0bin1 ELSE 0bin0 ENDIF";
+    }
+  }
+  else
+  {
+    throw "todo typecast2 "+op.type().id_string()+
+          " -> "+expr.type().id_string();
+  }
+}
+
+/*******************************************************************\
+
+Function: cvc_convt::convert_constant_expr
+
+  Inputs:
+
+ Outputs:
+
+ Purpose:
+
+\*******************************************************************/
+
+void cvc_convt::convert_constant_expr(const exprt &expr)
+{
+  if(expr.type().id()==ID_unsignedbv ||
+     expr.type().id()==ID_signedbv ||
+     expr.type().id()==ID_bv)
+  {
+    const irep_idt &value=expr.get(ID_value);
+
+    if(value.size()==8 ||
+       value.size()==16 ||
+       value.size()==32 ||
+       value.size()==64)
+    {
+      std::size_t w=value.size()/4;
+
+      mp_integer i=binary2integer(id2string(value), false);
+      std::string hex=integer2string(i, 16);
+
+      while(hex.size()<w) hex="0"+hex;
+
+      out << "0hex" << hex;
+    }
+    else
+    {
+      out << "0bin" << value;
+    }
+  }
+  else if(expr.type().id()==ID_pointer)
+  {
+    const irep_idt &value=expr.get(ID_value);
+
+    if(value=="NULL")
+    {
+      out << "(# object:="
+                   << pointer_logic.get_null_object()
+                   << ", offset:="
+                   << bin_zero(config.ansi_c.pointer_width) << " #)";
+    }
+    else
+      throw "unknown pointer constant: "+id2string(value);
+  }
+  else if(expr.type().id()==ID_bool)
+  {
+    if(expr.is_true())
+      out << "TRUE";
+    else if(expr.is_false())
+      out << "FALSE";
+    else
+      throw "unknown boolean constant";
+  }
+  else if(expr.type().id()==ID_array)
+  {
+    out << "ARRAY (i: " << array_index_type() << "):";
+
+    assert(!expr.operands().empty());
+
+    unsigned i=0;
+    forall_operands(it, expr)
```

Comment:
@tautschnig has suggested (https://github.com/diffblue/cbmc/pull/448#discussion_r96658825) that he would prefer these changes all done at once so I'll leave this for now.